src/main/java/dev/railroadide/railroad/java/JDKManager.java

```diff
@@ -0,0 +1,178 @@
+                            if (jdk != null) jdks.add(jdk);
+                        }
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        Map<String, JDK> uniqueJDKs = new LinkedHashMap<>();
+        for (JDK jdk : jdks) {
+            try {
+                String normalizedPath = Paths.get(jdk.path()).toRealPath().toString();
+                uniqueJDKs.putIfAbsent(normalizedPath, new JDK(normalizedPath, jdk.name(), jdk.version()));
+            } catch (IOException e) {
+                // fallback to raw path if normalization fails
+                uniqueJDKs.putIfAbsent(jdk.path(), jdk);
+            }
+        }
+
+        return new ArrayList<>(uniqueJDKs.values());
+    }
+
+
+    private static JavaVersion getJavaVersion(String javaHomeOrPath) {
+        String javaExecutable = javaHomeOrPath;
+        File f = new File(javaHomeOrPath, "bin/" + JAVA_EXECUTABLE);
+        if (f.exists() && f.canExecute()) {
+            javaExecutable = f.getAbsolutePath();
+        }
+
+        try {
+            Process p = new ProcessBuilder(javaExecutable, "-version")
+                .redirectErrorStream(true).start();
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+                String line;
+                Pattern pattern = Pattern.compile("\"(\\d+)\\.(\\d+).*\"");
+                while ((line = br.readLine()) != null) {
+                    Matcher matcher = pattern.matcher(line);
+                    if (matcher.find()) {
+                        int major = Integer.parseInt(matcher.group(1));
+                        int minor = Integer.parseInt(matcher.group(2));
+                        return new JavaVersion(major, minor);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            // Ignore
+        }
+        return null;
+    }
+
+    private static JDK createJDKFromJavaExecutable(String javaExecutablePath) {
+        File javaExe = new File(javaExecutablePath);
+        if (!javaExe.exists()) return null;
+
+        // Get home directory: parent of bin
+        File binDir = javaExe.getParentFile();
+        if (binDir == null) return null;
+        File javaHomeDir = binDir.getParentFile();
+        if (javaHomeDir == null) return null;
+
+        String javaHome = javaHomeDir.getAbsolutePath();
+        JavaVersion version = getJavaVersion(javaHome);
+        if (version != null) {
+            String name = javaHomeDir.getName(); // use folder name
+            return new JDK(javaHome, name, version);
+        }
+        return null;
+    }
+
+    private static String findJavaOnPath() {
+        String pathEnv = System.getenv("PATH");
+        if (pathEnv == null) return null;
+        String[] paths = pathEnv.split(File.pathSeparator);
```

Comment:
Just a minor formatting cleanup.
```suggestion
    private static String findJavaOnPath() {
        String pathEnv = System.getenv("PATH");
        if (pathEnv == null) 
            return null;

        String[] paths = pathEnv.split(File.pathSeparator);
```